subprojects/logging/src/main/java/org/gradle/internal/logging/slf4j/BuildOperationAwareLogger.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.slf4j;
+
+import org.gradle.api.logging.LogLevel;
+import org.gradle.api.logging.Logger;
+import org.gradle.internal.operations.OperationIdentifier;
+
+abstract class BuildOperationAwareLogger implements Logger {
```

Comment:
Why an abstract class rather than an interface?

Reply:
interface methods are public by default. I chose abstract class to allow the log method of that class to be package private only

Reply:
Got it, thanks